model/src/main/scala/org/broadinstitute/dsde/rawls/model/DataReferenceModel.scala

```diff
@@ -1,35 +1,85 @@
 package org.broadinstitute.dsde.rawls.model
 
-import bio.terra.workspace.model.{DataReferenceDescription, DataReferenceList}
+import java.util.UUID
+
+import bio.terra.workspace.model._
 import org.broadinstitute.dsde.workbench.model.{ValueObject, ValueObjectFormat}
 import spray.json.DefaultJsonProtocol._
+import spray.json.{DeserializationException, JsArray, JsObject, JsString, JsValue, RootJsonFormat, enrichAny}
 
 import scala.collection.JavaConverters._
 
 case class DataReferenceName(value: String) extends ValueObject
-case class DataRepoSnapshot(name: DataReferenceName, snapshotId: String)
-case class DataRepoSnapshotReference(referenceId: String, name: String, workspaceId: String, referenceType: Option[String], reference: Option[String], cloningInstructions: String)
-case class DataRepoSnapshotList(snapshots: List[DataRepoSnapshotReference])
-case class TerraDataRepoSnapshotRequest(instanceName: String, snapshot: String)
-
-object DataRepoSnapshotReference {
-  def apply(ref: DataReferenceDescription): DataRepoSnapshotReference = {
-    DataRepoSnapshotReference(ref.getReferenceId.toString, ref.getName, ref.getWorkspaceId.toString,
-      Option(ref.getReferenceType.toString), Option(ref.getReference), ref.getCloningInstructions.toString)
+case class NamedDataRepoSnapshot(name: DataReferenceName, snapshotId: String)
+
+object DataReferenceModelJsonSupport {
+  implicit object DataRepoSnapshotFormat extends RootJsonFormat[DataRepoSnapshot] {
+    val INSTANCE_NAME = "instanceName"
+    val SNAPSHOT = "snapshot"
+
+    def write(snap: DataRepoSnapshot) = JsObject(
+      INSTANCE_NAME -> JsString(snap.getInstanceName),
+      SNAPSHOT -> JsString(snap.getSnapshot)
+    )
+
+    def read(json: JsValue) = {
+      json.asJsObject.getFields(INSTANCE_NAME, SNAPSHOT) match {
+        case Seq(JsString(instanceName), JsString(snapshot)) =>
+          new DataRepoSnapshot().instanceName(instanceName).snapshot(snapshot)
+        case _ => throw new DeserializationException("DataRepoSnapshot expected")
+      }
+    }
```

Comment:
good call on using the spray-json de/serializers here, this keeps Rawls consistent in its approaches.

I am a bit wary about the lack of null handling. For writes, you might want to create a reusable method at the `DataReferenceModelJsonSupport ` object level:
```
  private def stringOrNull(in: String): JsValue = Option(in) match {
    case None => JsNull
    case Some(s) => JsString(s)
  }
```
and then your writes can take advantage of it:
```
    def write(snap: DataRepoSnapshot) = JsObject(
      INSTANCE_NAME -> stringOrNull(snap.getInstanceName),
      SNAPSHOT -> stringOrNull(snap.getSnapshot)
    )
```
you could extend `stringOrNull` to include `UUID`s or other classes that need serialization.

For reads, the concern is that if any of the json values are null, the `match` clauses won't match. For instance, if instanceName is null, then
```
json.asJsObject.getFields(INSTANCE_NAME, SNAPSHOT) match {
        case Seq(JsString(instanceName), JsString(snapshot)) =>
```
will not match, because you'll have a `Seq(JsNull, JsString)` instead of a `Seq(JsString, JsString)`. This might be what we WANT - to throw a `DeserializationException` if anything is null - but I want to make sure that's the intended behavior.

(this comment applies to all the new `*Format` classes in this PR)

Reply:
Added null handling for writing; I think we explicitly want to treat nulls as an error when reading, yes.